src/staking/manager.ts

```diff
@@ -0,0 +1,660 @@
+    if (inputUTXOs.length === 0) {
+      throw new Error("No input UTXOs provided");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const stakingPsbt = staking.toStakingPsbt(
+      unsignedStakingTx,
+      inputUTXOs,
+    );
+
+    const signedStakingPsbtHex = await this.btcProvider.signPsbt(
+      stakingPsbt.toHex()
+    );
+    return Psbt.fromHex(signedStakingPsbtHex).extractTransaction();
+  }
+
+  /**
+   * Creates a signed unbonding transaction that is ready to be sent to the BTC 
+   * network.
+   * @param stakingInput - The staking inputs.
+   * @param stakingParamVersion - The param version that was used to create the 
+   * EOI
+   * @param stakingTx - The staking transaction.
+   * @param unsignedUnbondingTx - The unsigned unbonding transaction.
+   * @param covenantUnbondingSignatures - The covenant unbonding signatures.
+   * @returns The signed unbonding transaction.
+   */
+  async createSignedBtcUnbondingTransaction(
+    stakingInput: StakingInputs,
+    stakingParamVersion: number,
+    stakingTx: Transaction,
+    unsignedUnbondingTx: Transaction,
+    covenantUnbondingSignatures: {
+      btcPkHex: string;
+      sigHex: string;
+    }[],
+  ) {
+    // Get the staking params at the time of the staking transaction
+    const p = getBbnParamByVersion(stakingParamVersion, this.stakingParams);
+    if (!p) {
+      throw new Error(
+        `Unable to find staking params for version ${stakingParamVersion}`,
+      );
+    }
+
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error(
+        "Staker info not found while creating signed unbonding transaction",
+      );
+    }
+
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const unbondingPsbt = staking.toUnbondingPsbt(
+      unsignedUnbondingTx,
+      stakingTx,
+    );
+    const signedUnbondingPsbtHex = await this.btcProvider.signPsbt(
+      unbondingPsbt.toHex()
+    );
+    const signedUnbondingTx = Psbt
+      .fromHex(signedUnbondingPsbtHex).extractTransaction();
+
+    // Add covenant unbonding signatures
+    // Convert the params of covenants to buffer
+    const covenantBuffers = p.covenantNoCoordPks.map((covenant) =>
+      Buffer.from(covenant, "hex"),
+    );
+    const witness = createCovenantWitness(
+      signedUnbondingTx.ins[0].witness,
+      covenantBuffers,
+      covenantUnbondingSignatures,
+      p.covenantQuorum,
+    );
+    // Overwrite the witness to include the covenant unbonding signatures
+    signedUnbondingTx.ins[0].witness = witness;
+
+    return signedUnbondingTx;
+  }
+
+  /**
+   * Creates a signed withdrawal transaction on the unbonding path that is 
```

Comment:
hmm unbonding output does not have unbonding path, so maybe something like:
```
signed withdrawal transaction on the unbodning output expiry path 
```
?